### ISSUE TITLE
基于传统聚类算法的cluster-guided训练完整实现（含主流程代码和可视化）

### ISSUE THREAD

**Issue by Xiyu-9** [Xiyu-9/first#2]
## 背景
我们希望用传统的聚类算法（K-means、Spectral Clustering、Agglomerative Clustering）直接指导模型聚类，替换掉原有的对比/聚类损失函数，并对比三种算法下的聚类表现。

## 需求
1. **重构训练流程**：
   - 移除所有自定义loss（如SupConLoss, ClusterLoss等）。
   - 训练过程中，不再基于标签或对比损失，改为：
     1. 每N个epoch，用当前`instance_projector(gs1)`输出的嵌入，对所有训练样本进行聚类（选择K-means、SC、AC之一，参数可配置）。
     2. 把聚类结果作为"伪标签"，用普通的交叉熵loss训练网络（对`instance_projector(gs1)`输出）。
     3. 重复若干轮，更新伪标签和网络参数。

2. **指标与可视化**：
   - 在每种聚类算法下，分别评估：
     - Clustering Accuracy (ACC)
     - Silhouette Score (Sil.)
     - Inter-cluster Separation (Sep.)
   - 训练结束后，输出每种算法下的t-SNE聚类可视化（和原论文一致，21类不同颜色，中心点标注）。

3. **代码结构建议**：
   - 增加参数`--clustering_method`，支持选择`kmeans`、`spectral`、`agglomerative`。
   - 主训练流程可参考如下伪代码：
     ```python
     for epoch in range(epochs):
         # 提取全部训练数据嵌入
         embeddings = get_all_embeddings(train_loader, encoder_model)
         # 用选定聚类算法聚类，获得伪标签
         pseudo_labels = clustering(embeddings, method=args.clustering_method)
         # 按伪标签训练模型
         for data, plabel in zip(train_loader, pseudo_labels):
             ... # 交叉熵loss
     ```
   - 可以把主要新代码写在`cluster_guided_train`函数里。

4. **可复现性**
   - 给出完整的、基于当前gclc.py重构后的、可独立运行的main流程代码（可直接替换原main）。
   - 明确说明如何切换聚类方法/参数。

## 附加说明
- 只用`instance_projector(gs1)`做聚类与训练，忽略所有cluster_projector相关内容。
- 可参考并复用原gclc.py的数据处理、模型结构部分。
- 可兼容原有命令行参数。

## 交付内容
- 完整的主代码（建议命名为`gclc_cluster_guided.py`），可直接运行。
- 详细注释说明每个关键步骤。
- 训练和评估流程示例。
- 聚类和可视化部分示例。

---

如需示意图片，请参考下面的t-SNE可视化效果：

<img alt="Instance-t-SNE Visualization with Class Centroids" width="1400" src="https://github.com/user-attachments/assets/6e488ac5-679e-40db-92a9-05e318e688ce" />
